### PR TITLE
1985 bulk delete

### LINKED ISSUE #2081: Create bulk delete functionality in web front end
This is a necessary step before continuing on #1985

### LINKED ISSUE #2082: Create bulk delete resolver
This is necessary for completing #1985 and relates to #2081

### LINKED ISSUE #2083: Create bulk delete function on datalad-server interface
Maybe will be directly in datalad, this area seems to be the source of the bugs.

This is necessary for completing #1985 and relates to #2081 and #2082

### PR DESCRIPTION
**Pull request by tranttommy**
Tickets addressed:
resolves #2081, resolves #2082, resolves #2083 as part of #1985

### CODE REVIEW

File: services/datalad/datalad_service/handlers/files.py [hunk 1]
```diff
@@ -131,6 +132,24 @@ def on_delete(self, req, resp, dataset, filename):
...
+            name, email = get_user_info(req)
+            if name and email:
+                media_dict['name'] = name
+                media_dict['email'] = email
+            try:
+                remove_files(self.store, dataset, files=deleted_files, name=name, email=email, cookies=req.cookies)
+                resp.media = media_dict
+                resp.status = falcon.HTTP_OK
+            except:
+                resp.status = falcon.HTTP_INTERNAL_SERVER_ERROR
```
Comment (nellh) [inline services/datalad/datalad_service/handlers/files.py]: The implementation here looks fine but I wouldn't overload the same API now three paths to doing the same thing. If this is its own resource handler it wouldn't need the req.bounded_stream hack. req.media does this internally when accessed, including the JSON parsing.

This should also have test coverage.

File: packages/openneuro-server/src/graphql/schema.js [hunk 2]
```diff
@@ -161,6 +161,13 @@ export const typeDefs = `
     reexportRemotes(datasetId: ID!): Boolean
     # Reset draft commit
     resetDraft(datasetId: ID!, ref: String!): Boolean
+    # Bulk delete individual files
+    deleteBulk(datasetId: ID!, files: [DeleteFile]): Boolean
```
Comment (nellh) [inline packages/openneuro-server/src/graphql/schema.js]: Can we consolidate the delete APIs down to one deleteFiles API? The other cases map pretty well to this implementation.

File: packages/openneuro-server/src/graphql/resolvers/dataset.js [hunk 3]
```diff
@@ -181,6 +181,35 @@ export const deleteFile = async (
...
+  { datasetId, files },
+  { user, userInfo },
+) => {
+  try {
+    await checkDatasetWrite(datasetId, user, userInfo)
+    const deletedFiles = await datalad
+      .deleteBulk(datasetId, files, userInfo)
+      .then(filenames =>
+        Promise.all(filenames.map(filename => new UpdatedFile(filename))),
+      )
```
Comment (nellh) [inline packages/openneuro-server/src/graphql/resolvers/dataset.js:142]: Looks like this can fail silently if you delete files that don't exist?

**Comment by nellh** [comment]
This looks good except the styling needs some tweaks. Maybe we could just put the bulk delete button in the top level actions and replace the checkbox with a toggle kind of like the star/follow buttons as a quick improvement?

image

**Comment by tranttommy** [comment]
I believe @thinknoack will be taking care of the styling

**Pull request merged**